pkg/unstruct/v1alpha2/kubernetes.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"context"
+	"strings"
+
+	kclient "github.com/openebs/maya/pkg/kubernetes/client/v1alpha2"
+	provider "github.com/openebs/maya/pkg/provider/v1alpha1"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// kubeclient enables kubernetes API operations on catalog
+// instance
+type kubeclient struct {
+	client.Client  // kubernetes client
+	kclient.Handle // manage kubernetes client & enable mocking
+}
+
+// kubeclientBuildOption defines the abstraction to build
+// a kubeclient instance
+type kubeclientBuildOption func(*kubeclient)
+
+func withDefaults(k *kubeclient) error {
+	if k.Client == nil {
+		cli, err := kclient.New()
+		if err != nil {
+			return err
+		}
+		k.Client = cli
+	}
+	return nil
+}
+
+// WithKubeClient sets the kubernetes client against
+// the kubeclient instance
+func WithKubeClient(c client.Client) kubeclientBuildOption {
```

Comment:
Let us rename this to `WithClient` else it is becoming repetitive with `KubeClient`

Reply:
done